tests/wfcancel.test.ts

```diff
@@ -0,0 +1,164 @@
+import { StatusString, DBOS } from '../src';
+import { DBOSConfig } from '../src/dbos-executor';
+import { generateDBOSTestConfig, setUpDBOSTestDb } from './helpers';
+import { v4 as uuidv4 } from 'uuid';
+
+describe('wf-cancel-tests', () => {
+  let config: DBOSConfig;
+
+  beforeAll(async () => {
+    config = generateDBOSTestConfig();
+    await setUpDBOSTestDb(config);
+    DBOS.setConfig(config);
+  });
+
+  beforeEach(async () => {
+    WFwith2Steps.stepsExecuted = 0;
+    await DBOS.launch();
+  });
+
+  afterEach(async () => {
+    await DBOS.shutdown();
+  }, 10000);
+
+  test('test-two-steps-base', async () => {
+    const wfid = uuidv4();
+    const wfh = await DBOS.startWorkflow(WFwith2Steps, { workflowID: wfid }).workflowWithSteps();
+
+    await wfh.getResult();
+
+    expect(WFwith2Steps.stepsExecuted).toBe(2);
+  });
+
+  test('test-two-steps-cancel', async () => {
+    const wfid = uuidv4();
+
+    try {
+      const wfh = await DBOS.startWorkflow(WFwith2Steps, { workflowID: wfid }).workflowWithSteps();
+
+      await DBOS.executor.cancelWorkflow(wfid);
+      await wfh.getResult();
```

Comment:
Assert false after the `getResult`, to ensure the call succeeding fails the test